### ISSUE TITLE
easywave SYCL performance

### ISSUE THREAD

**Issue by ghost** [christgau/easywave-sycl#1]
Greetings @christgau 

I am wondering if I can make a contribution or code changes to your easywave-sycl version? We have a performant easywave-SYCL version from our [Velocity Bench](https://github.com/oneapi-src/Velocity-Bench/tree/main/easywave), which we removed the DPCT functions, especially the low performing 2-D `memcpy` function. The functionality between the two versions are the same.

**Comment by christgau** [comment]
@jgtong. Sure, contributions are welcome. Just provide a PR and I will take a look at it.

**Comment by ghost** [comment]
@christgau : May I have push access to this repo? It says that I do not have permission

**Comment by christgau** [comment]
That's right, because that's different from creating a pull request if I understand you correctly.

From my point of view: Please create a fork first if you haven't done yet and then create a PR to this repo. Direct push (into master/main) is not the (my) preferred way to "merge" contrbutions.
